sdks/java/harness/src/test/java/org/apache/beam/fn/harness/FnApiDoFnRunnerTest.java

```diff
@@ -133,3333 +138,4296 @@
-                    KV.of("5", KV.of(new OffsetRange(0, 2), GlobalWindow.TIMESTAMP_MIN_VALUE)),
-                    2.0)),
-            valueInGlobalWindow(
-                KV.of(
-                    KV.of("2", KV.of(new OffsetRange(0, 1), GlobalWindow.TIMESTAMP_MIN_VALUE)),
-                    1.0))));
-    mainOutputValues.clear();
+      FnDataReceiver<WindowedValue<?>> mainInput =
+          consumers.getMultiplexingConsumer(inputPCollectionId);
+      IntervalWindow window1 = new IntervalWindow(new Instant(5), new Instant(10));
+      IntervalWindow window2 = new IntervalWindow(new Instant(6), new Instant(11));
+      WindowedValue<?> firstValue = valueInWindows("5", window1, window2);
+      WindowedValue<?> secondValue = valueInWindows("2", window1, window2);
+      mainInput.accept(firstValue);
+      mainInput.accept(secondValue);
+      assertThat(
+          mainOutputValues,
+          contains(
+              WindowedValue.of(
+                  KV.of("5", KV.of(new OffsetRange(0, 5), GlobalWindow.TIMESTAMP_MIN_VALUE)),
+                  firstValue.getTimestamp(),
+                  window1,
+                  firstValue.getPane()),
+              WindowedValue.of(
+                  KV.of("5", KV.of(new OffsetRange(0, 5), GlobalWindow.TIMESTAMP_MIN_VALUE)),
+                  firstValue.getTimestamp(),
+                  window2,
+                  firstValue.getPane()),
+              WindowedValue.of(
+                  KV.of("2", KV.of(new OffsetRange(0, 2), GlobalWindow.TIMESTAMP_MIN_VALUE)),
+                  secondValue.getTimestamp(),
+                  window1,
+                  secondValue.getPane()),
+              WindowedValue.of(
+                  KV.of("2", KV.of(new OffsetRange(0, 2), GlobalWindow.TIMESTAMP_MIN_VALUE)),
+                  secondValue.getTimestamp(),
+                  window2,
+                  secondValue.getPane())));
+      mainOutputValues.clear();
 
-    assertTrue(finishFunctionRegistry.getFunctions().isEmpty());
-    assertThat(mainOutputValues, empty());
+      assertTrue(finishFunctionRegistry.getFunctions().isEmpty());
+      assertThat(mainOutputValues, empty());
 
-    Iterables.getOnlyElement(teardownFunctions).run();
-    assertThat(mainOutputValues, empty());
-  }
+      Iterables.getOnlyElement(teardownFunctions).run();
+      assertThat(mainOutputValues, empty());
+    }
 
-  @Test
-  public void testProcessElementForWindowedTruncateAndSizeRestriction() throws Exception {
-    Pipeline p = Pipeline.create();
-    PCollection<String> valuePCollection = p.apply(Create.of("unused"));
-    PCollectionView<String> singletonSideInputView = valuePCollection.apply(View.asSingleton());
-    valuePCollection
-        .apply(Window.into(SlidingWindows.of(Duration.standardSeconds(1))))
-        .apply(
-            TEST_TRANSFORM_ID,
-            ParDo.of(new WindowObservingTestSplittableDoFn(singletonSideInputView))
-                .withSideInputs(singletonSideI
```

Comment:
Explanation on why in python test it's 64 but here is 63:
Java uses `BigDecimal` and result here is `63.9xxxxx`. When Java performs `BigDecimal.longValue()`, it wil round down to `63`.


`